v2/test/postgresql_test.go

```diff
@@ -0,0 +1,377 @@
+
+	// Confirm we can connect as the user
+	conn, err = postgresqlutil.ConnectToDB(
+		tc.Ctx,
+		fqdn,
+		"",
+		postgresqlutil.PSqlServerPort,
+		user.Spec.AzureName,
+		password)
+	tc.Expect(err).ToNot(HaveOccurred())
+	// Close the connection
+	tc.Expect(conn.Close()).To(Succeed())
+
+	// Update the secret
+	newPassword := tc.Namer.GeneratePassword()
+	newSecret := &v1.Secret{
+		ObjectMeta: userSecret.ObjectMeta,
+		StringData: map[string]string{
+			passwordKey: newPassword,
+		},
+	}
+	tc.UpdateResource(newSecret)
+
+	// Connect to the DB as the user, using the new secret
+	tc.G.Eventually(
+		func() error {
+			conn, err = postgresqlutil.ConnectToDB(
+				tc.Ctx,
+				fqdn,
+				"",
+				postgresqlutil.PSqlServerPort,
+				user.Spec.AzureName,
+				newPassword)
+			if err != nil {
+				return err
+			}
+
+			return conn.Close()
+		},
+		2*time.Minute, // We expect this to pass pretty quickly
+	).Should(Succeed())
+
+	tc.DeleteResourceAndWait(user)
+}
+
+func Test_PostgreSql_User(t *testing.T) {
+	t.Parallel()
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	adminUsername := "myadmin"
+	adminPasswordKey := "adminPassword"
+	adminPassword := tc.Namer.GeneratePassword()
+	adminSecret := newPostgresSqlSecret(tc, adminPasswordKey, adminPassword)
+
+	passwordKey := "password"
+	password := tc.Namer.GeneratePassword()
+	userSecret := newPostgresSqlSecret(tc, passwordKey, password)
+
+	tc.CreateResource(adminSecret)
+	tc.CreateResource(userSecret)
+
+	flexibleServer := newPostgreSqlServer(tc, rg, adminUsername, adminPasswordKey, adminSecret.Name)
+	firewallRule := newPostgreSqlServerOpenFirewallRule(tc, flexibleServer)
+
+	user := &postgresqlv1.User{
+		ObjectMeta: tc.MakeObjectMetaWithName(tc.NoSpaceNamer.GenerateName("user")),
+		Spec: postgresqlv1.UserSpec{
+			Owner: testcommon.AsOwner(flexibleServer),
+			Roles: []string{
+				"azure_pg_admin",
+			},
+			LocalUser: &postgresqlv1.LocalUserSpec{
+				ServerAdminUsername: adminUsername,
+				ServerAdminPassword: flexibleServer.Spec.AdministratorLoginPassword,
+				Password: &genruntime.SecretReference{
+					Name: userSecret.Name,
+					Key:  passwordKey,
+				},
+			},
+		},
+	}
+	tc.CreateResourcesAndWait(flexibleServer, firewallRule, user)
+
+	tc.DeleteResourceAndWait(user)
+}
+
+func newPostgresSqlSecret(tc *testcommon.KubePerTestContext, key string, password string) *v1.Secret {
+	secret := &v1.Secret{
+		ObjectMeta: tc.MakeObjectMeta("postgresqlsecret"),
+		StringData: map[string]string{
+			key: password,
+		},
+	}
+
+	return secret
+}
+
+func newPostgreSqlServer(tc *testcommon.KubePerTestContext, rg *resources.ResourceGroup, adminUsername string, adminKey string, adminSecretName string) *postgresql.FlexibleServer {
+	// Force this test to run in a region that is not capacity constrained.
+	// location := tc.AzureRegion TODO: Uncomment this line when West US 2 is no longer constrained
+	location := to.Ptr("australiaeast")
```

Comment:
minor: You can do this at the top of the test (line ~29) by doing:
`tc.AzureRegion = to.Ptr("australiaeast")`

Then use `tc.AzureRegion` in all of the helpers.

The advantage is that ensures even if you had supporting resources they would get provisioned alongside it, and it also moves the hardcoding up to the very top of the test so it's clear to subsequent test runners that's where they need to change it if in the future restrictions change and the region that used to work doesn't anymore.

Reply:
Done